test/foundry/unit/concrete/erc4337account/TestERC4337Account_OnlyEntryPoint.t.sol

```diff
@@ -20,21 +21,37 @@ contract TestERC4337Account_OnlyEntryPoint is NexusTest_Base {
     /// @notice Verifies that a valid operation passes validation when invoked from the EntryPoint.
     function test_ValidUserOpFromEntryPoint() public {
         PackedUserOperation[] memory userOps = new PackedUserOperation[](1);
-        userOps[0] = buildPackedUserOp(BOB.addr, getNonce(address(BOB_ACCOUNT), address(VALIDATOR_MODULE)));
+        userOps[0] = buildPackedUserOp(userAddress, getNonce(address(BOB_ACCOUNT), MODE_VALIDATION, address(VALIDATOR_MODULE)));
         bytes32 userOpHash = ENTRYPOINT.getUserOpHash(userOps[0]);
         userOps[0].signature = signMessage(BOB, userOpHash); // Sign operation with valid signer
 
         startPrank(address(ENTRYPOINT));
+        // Attempt to validate the user operation, expecting success
+        uint256 res = BOB_ACCOUNT.validateUserOp(userOps[0], userOpHash, 10);
+        assertTrue(res == 0, "Valid operation should pass validation");
+        stopPrank();
+    }
+
+    function test_ValidateUserOp_InvalidSignature() public {
+        startPrank(address(ENTRYPOINT));
+        // Initialize a user operation with a valid nonce but signed by an incorrect signer
+        PackedUserOperation[] memory userOps = new PackedUserOperation[](1);
+        userOps[0] = buildPackedUserOp(userAddress, getNonce(address(BOB_ACCOUNT), MODE_VALIDATION, address(VALIDATOR_MODULE)));
+        bytes32 userOpHash = ENTRYPOINT.getUserOpHash(userOps[0]);
+        userOps[0].signature = signMessage(ALICE, userOpHash); // Incorrect signer simulated
+
+        startPrank(address(ENTRYPOINT));
+        // Attempt to validate the user operation, expecting failure due to invalid signature
         uint256 res = BOB_ACCOUNT.validateUserOp(userOps[0], userOpHash, 0);
         stopPrank();
 
-        assertTrue(res == 0, "Valid operation should pass validation");
+        assertTrue(res == 1, "Invalid operation should not pass validation");
     }
 
     /// @notice Ensures that operations fail validation when invoked from an unauthorized sender.
     function test_RevertIf_UserOpFromNonEntryPoint() public {
         PackedUserOperation[] memory userOps = new PackedUserOperation[](1);
-        userOps[0] = buildPackedUserOp(userAddress, getNonce(address(BOB_ACCOUNT), address(VALIDATOR_MODULE)));
+        userOps[0] = buildPackedUserOp(userAddress, getNonce(address(BOB_ACCOUNT), MODE_VALIDATION, address(VALIDATOR_MODULE)));
```

Comment:
what kind of other modes could be there?

Reply:
idk yet, maybe MODE_DEFAULT if decide to introduce at some point